pkg/component/etcd/lease/lease.go

```diff
@@ -91,25 +89,15 @@ func New(c client.Client, logger logr.Logger, namespace string, values Values) I
 }
 
 func (c *component) emptyLease(name string) *coordinationv1.Lease {
-	labels := map[string]string{"name": "etcd", "instance": c.values.EtcdName}
+	copyLabels := make(map[string]string, len(c.values.Labels))
+	for k, v := range c.values.Labels {
+		copyLabels[k] = v
+	}
```

Comment:
We utilize labels in the two leases `member` and `snapshot`. Given the mutable nature of a `map` in Golang, if we assign `c.values.Labels` directly to a lease's metadata, any modifications to one lease's metadata would inadvertently impact the other. Notably, `member` lease labels are updated [here](https://github.com/gardener/etcd-druid/blob/ea9f5ae5a4fa6e18a3e955b15632d40573175489/pkg/component/etcd/lease/lease_member.go#L41-L58), which would undesirably affect the `snapshot` lease. To avoid this, we are making a copy of the labels.

Reply:
Ok. Then please use golang's in-built [`copy`](https://golangbyexample.com/copy-function-in-golang/) function for this.

Reply:
In-built `copy` works only for slice 😢